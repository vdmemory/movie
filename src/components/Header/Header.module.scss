.header {
    outline: 1px solid green;
}