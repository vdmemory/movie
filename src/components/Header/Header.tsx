import React, { useEffect } from 'react'

import styles from "./Header.module.scss";
import { useDispatch, useSelector } from 'react-redux';
import { selectSearchMovies, selectDetailsMovie } from '../../store/selectors';
import { getSearchMoviesRequest, getDetailsMovieRequest } from '../../store/actions';

const Header: React.FC = () => {
    const dispatch = useDispatch();

    const {searchMoviesData, searchMoviesLoading, searchMoviesError} = useSelector(selectSearchMovies);

    const {detailsMovieData} = useSelector(selectDetailsMovie)

    console.log(searchMoviesData);

    useEffect(() => {
        dispatch(getSearchMoviesRequest('hgvfghvfytfytvytyt'))
        dispatch(getDetailsMovieRequest("tt0093692"));
    }, [dispatch])

    return <div className={styles.header}>Header</div>;
};

export default Header;
