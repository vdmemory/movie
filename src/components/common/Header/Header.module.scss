.header {
    padding: 15px;
    background: aliceblue;

    & .title {
        font-family: 'Roboto';
        font-size: 22px;
        text-align: center;
    }
}





