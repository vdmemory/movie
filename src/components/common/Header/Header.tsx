import React, { Dispatch } from "react";
import Search from "../Search/Search";

import styles from "./Header.module.scss";

type Props = {
    placeholder: string;
    titleHeader: string;
    titleSearch: string;
    type: string;
    dispatch: Dispatch<any>;
};

const Header: React.FC<Props> = ({
    titleHeader,
    titleSearch,
    type,
    placeholder,
    dispatch,
}) => {
    return (
        <header className={styles.header}>
            <h1 className={styles.title}>{titleHeader}</h1>
            <Search
                placeholder={placeholder}
                title={titleSearch}
                type={type}
                dispatch={dispatch}
            />
        </header>
    );
};

export default Header;
