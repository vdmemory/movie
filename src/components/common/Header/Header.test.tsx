import React, { Dispatch } from "react";
import { createRenderer } from "react-test-renderer/shallow";

import Header from "./Header";

type Props = {
    placeholder: string;
    titleHeader: string;
    titleSearch: string;
    type: string;
    dispatch: Dispatch<any>;
};

const setup = (props: Props) => {
    const renderer = createRenderer();
    renderer.render(<Header {...props} />);
    const output = renderer.getRenderOutput();
    return { output, props };
};

const propsRender: Props = {
    placeholder: "this placeholder text",
    titleHeader: "this title header",
    titleSearch: "this title search",
    type: "this type from input search",
    dispatch: jest.fn(),
};

describe("Header Component", () => {
    describe("Render component and its components", () => {
        test("Should renders correctly", () => {
            const { output } = setup(propsRender);
            expect(output).toMatchSnapshot();
        });

        test("Should render container and has a header class", () => {
            const { output } = setup(propsRender);
            expect(output.type).toBe("header");
            expect(output.props.className).toBe("header");
        });

        test("Should render title header and has a title class and text", () => {
            const { output } = setup(propsRender);
            const [title] = output.props.children;
            expect(title.type).toBe("h1");
            expect(title.props.className).toBe("title");
            expect(title.props.children).toBe("this title header");
        });

    });
});
