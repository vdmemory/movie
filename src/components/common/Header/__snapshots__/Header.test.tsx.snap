// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Render component and its components Should renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    this title header
  </h1>
  <Search
    dispatch={[MockFunction]}
    placeholder="this placeholder text"
    title="this title search"
    type="this type from input search"
  />
</header>
`;
